jmespath-core/src/main/java/io/burt/jmespath/BaseAdapter.java

```diff
@@ -60,8 +58,8 @@ public int compare(T value1, T value2) {
         case BOOLEAN:
           return (isTruthy(value1) && isTruthy(value2)) || (!isTruthy(value1) && !isTruthy(value2)) ? 0 : -1;
         case NUMBER:
-          Comparable d1 = toNumber(value1).doubleValue();
-          Comparable d2 = toNumber(value2).doubleValue();
+          Double d1 = toNumber(value1).doubleValue();
+          Double d2 = toNumber(value2).doubleValue();
           return d1.compareTo(d2);
```

Comment:
Actually that quote from the JMESPath docs was just for comparisons, not for sorting, and this code is used for both – so we're good. The docs for `sort` are inconsistent, they say that `sort` takes either an array of numbers or an array of strings, but then goes on to show what happens when you sort `[false, [], null]`… so I'm not sure what's what.